src/librustc_lexer/src/utils.rs

```diff
@@ -0,0 +1,72 @@
+//! Helper functions for parsing Rust syntax.
```

Comment:
Well, I thought that they'll fit better in the separate module since they don't actually either produce tokens or affecting the processed stream in some way, and you don't really have to look at their implementation to understand the `lexer` module itself.

Usually I try to keep in the crate root only things that are really a "heart" of the module and required to be understood, which is not the case for those functions.

Reply:
They key here is that these functions themselves are `pub`, so they are just as important part of public API as the `Token`.

Reply:
Well, yeah, and I think that they all can be described as "lexer helper utilities", don't they? Of course I don't insist and can revert them, but I don't really think that those changes decrease overall readability of the crate.